### PR TITLE
init descriptor value if not contains key

### PR DESCRIPTION
**Pull request by groovelab**
Run as GATT Server that has notify characteristic.
After connected from central, the sequences from GATT client are service discovery and subscribe notify characteristic.
However, the sequences from some GATT client are like this.

1. service discovery
2. subscribe notify characteristic
3. service discovery again

I don't know why some GATT client do service discovery again.
That may be GATT client bug.

But, I want to prevent initialize descriptor values from 2nd service discovery.
In other words, I want to keep characteristic subscribe status.

### CODE REVIEW

**Comment by philips77** [comment]
I was addressing this double service discovery issue recently on *develop* branch. I'll be releasing the new version soon.

**Comment by groovelab** [comment]
I see. Close this PR.
